<launch>
  <!-- Let Mission Execution know we are running the script in simulation -->
  <param name="real_map" value="false"/>
  
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [waffle, waffle_pi]"/>
  <arg name="gui" default="false"/>
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="x_pos" default="0.25"/>
  <arg name="y_pos" default="0.25"/>
  <arg name="z_pos" default="0.0"/>
  
  <rosparam file="$(find turtlebot3_manipulation_gazebo)/config/gazebo_controller.yaml" command="load"/>
  
  <!-- send robot urdf to param server -->
  <include file="$(find turtlebot3_manipulation_description)/launch/turtlebot3_manipulation_upload.launch">
    <arg name="model" value="$(arg model)"/>
  </include>
  
  <!-- Launch world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find assignment4_ttk4192)/world/turtlebot3_ttk4192.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>
  
  <!-- Spawn robot and manipulator -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-urdf -model turtlebot3_manipulation
              -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos)
              -param robot_description
              -J joint1 0.0 -J joint2 0.0 -J joint3 0.0 -J joint4 0.0 -J gripper 0.0 -J gripper_sub 0.0"/>
              
  <!-- controller utils -->
  <include file="$(find turtlebot3_manipulation_gazebo)/launch/controller_utils.launch"/>

  <!-- run controllers -->
  <include file="$(find turtlebot3_manipulation_gazebo)/launch/turtlebot3_manipulation_controller.launch"/>
              
  <!-- Launch MoveIt move_group -->
  <param name="use_sim_time" value="true"/>
  <include file="$(find turtlebot3_manipulation_moveit_config)/launch/move_group.launch"/>
  
  <!-- Launch Manipulator GUI-->
  <!-- <include file="$(find turtlebot3_manipulation_gui)/launch/turtlebot3_manipulation_gui.launch"/> -->
 
</launch>